In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [2]:
data_1 = pd.read_csv('E:/SATYAJEET/Assignments/Assignment 3/Costomer+OrderForm.csv')
data_1.describe

<bound method NDFrame.describe of     Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]>

In [4]:
data_1.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [5]:
data_1.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [6]:
data_1.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [7]:
data_1.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [8]:
data_2 = np.array([[271,267,269,280],[29,33,31,20]])

In [9]:
data_3 = chi2_contingency(data_2)

In [10]:
data_3

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [11]:
data_3[1]

0.2771020991233135

#### Comparing values of p value = 0.2771020991233135 and ɑ = 0.05
#### this implicates that we should consider null hupothesis as p value > ɑ